// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package privacy_sandbox.server_common.gvisor;

service RomaGvisorService {
  rpc LoadBinary(LoadBinaryRequest) returns (LoadBinaryResponse) {}
  rpc ExecuteBinary(ExecuteBinaryRequest) returns (ExecuteBinaryResponse) {}
  rpc ExecuteBinaryBidiStreaming(stream ExecuteBinaryBidiStreamingRequest)
      returns (stream ExecuteBinaryBidiStreamingResponse) {}
}

message ExecuteBinaryRequest {
  bytes code_token = 1;
  // Serialized request proto defined by the integrator
  bytes serialized_request = 2;
}

message ExecuteBinaryResponse {
  // Serialized response proto defined by the integrator
  bytes serialized_response = 1;
}

message ExecuteBinaryBidiStreamingRequest {
  bytes code_token = 1;
  // Serialized request proto defined by the integrator
  bytes serialized_request = 2;
}

message ExecuteBinaryBidiStreamingResponse {
  // Serialized response proto defined by the integrator
  bytes serialized_response = 1;
}

message LoadBinaryRequest {
  bytes code_token = 1;
  bytes code = 2;
}

message LoadBinaryResponse {
  bytes code_token = 1;
}
